### PR TITLE
Mkulakow/kfs rest ut

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/test/kfs_rest_parser_test.cpp [hunk 1]
```diff
@@ -0,0 +1,769 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
```
Comment (atobiszei) [inline src/test/kfs_rest_parser_test.cpp:2]: Copyright year

File: src/test/kfs_rest_parser_test.cpp [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+}
+
+TEST_F(KFSRestParserTest, parseValidRequestWithNoDataButBinaryInputsParameter) {
+    std::string request = R"({
+    "inputs" : [
+        {
+        "name" : "input0",
+        "shape" : [ 2, 2 ],
+        "datatype" : "BYTES",
+        "parameters" : {"binary_data_size" : 16}
```
Comment (mzegla) [inline src/test/kfs_rest_parser_test.cpp:599]: it's irrelevant from the parser perspective, but it looks like this size does not match shape and datatype

File: src/test/kfs_rest_parser_test.cpp [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+    auto status = parser.parse(request.c_str());
+    ASSERT_EQ(status, StatusCode::OK);
+
+    auto proto = parser.getProto();
+    ASSERT_EQ(proto.inputs_size(), 1);
+    ASSERT_EQ(proto.inputs()[0].name(), "input0");
+    ASSERT_THAT(proto.inputs()[0].shape(), ElementsAre(2, 2));
+    ASSERT_EQ(proto.inputs()[0].datatype(), "FP64");
+    ASSERT_EQ(proto.inputs()[0].contents().fp64_contents_size(), 4);
+    ASSERT_THAT(proto.inputs()[0].contents().fp64_contents(), ElementsAre((float)1.1, (float)2.2, (float)3.3, (float)4.4));
```
Comment (atobiszei) [inline src/test/kfs_rest_parser_test.cpp:230]:
```suggestion
    ASSERT_THAT(proto.inputs()[0].contents().fp64_contents(), ElementsAre(1.1f, 2.2f, 3.3f, 4.4f));
```

File: src/test/kfs_rest_parser_test.cpp [hunk 4]
```diff
@@ -0,0 +1,769 @@
...
+        "shape" : [ 2, 2 ],
+        "datatype" : "UINT32"
+        }
+    ]
+    })";
+    auto status = parser.parse(request.c_str());
+    ASSERT_EQ(status, StatusCode::REST_COULD_NOT_PARSE_INPUT);
+}
+
+TEST_F(KFSRestParserTest, parseInvalidRequestWithDataAndBYTESdatatype) {
```
Comment (mzegla) [inline src/test/kfs_rest_parser_test.cpp:678]: This scenario is explicitly disallowed in our implementation?

File: src/test/kfs_rest_test.cpp [hunk 5]
```diff
@@ -185,6 +185,20 @@ TEST_F(HttpRestApiHandlerTest, RegexParseServerLive) {
     ASSERT_EQ(comp.type, ovms::KFS_GetServerLive);
 }
 
+TEST_F(HttpRestApiHandlerTest, RegexParseInferWithBinaryInputs) {
+    std::string request = "/v2/models/dummy/versions/1/infer";
```
Comment (mzegla) [inline src/test/kfs_rest_test.cpp:189]: Do we also have a test when version is not specified?

**Review by atobiszei — APPROVED** [review]

**Review by mzegla — APPROVED** [review]

**Pull request merged**